deps/jerry/jerry-ext/include/jerryscript-ext/handle-scope.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_HANDLE_SCOPE_H
+#define JERRYX_HANDLE_SCOPE_H
+
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+#ifndef JERRY_X_HANDLE_SCOPE_PRELIST_HANDLE_COUNT
+#define JERRY_X_HANDLE_SCOPE_PRELIST_HANDLE_COUNT 20
+#endif
+
+#ifndef JERRY_X_HANDLE_SCOPE_PRELIST_SCOPE_COUNT
+#define JERRY_X_HANDLE_SCOPE_PRELIST_SCOPE_COUNT 20
+#endif
+
+typedef struct jerryx_handle_t jerryx_handle_t;
+struct jerryx_handle_t {
+  jerry_value_t jval;
+  jerryx_handle_t *sibling;
+};
+
+typedef struct jerryx_handle_scope_t jerryx_handle_scope_t;
+typedef jerryx_handle_scope_t *jerryx_handle_scope;
+typedef jerryx_handle_scope_t *jerryx_escapable_handle_scope;
+struct jerryx_handle_scope_t {
+  jerry_value_t handle_prelist[JERRY_X_HANDLE_SCOPE_PRELIST_HANDLE_COUNT];
+  size_t handle_count;
+  jerryx_handle_t *handle_ptr;
+};
+
+
+typedef struct jerryx_handle_scope_dynamic_t jerryx_handle_scope_dynamic_t;
+struct jerryx_handle_scope_dynamic_t {
+  jerry_value_t handle_prelist[JERRY_X_HANDLE_SCOPE_PRELIST_HANDLE_COUNT];
+  size_t handle_count;
```

Comment:
To improve the code readability, see https://github.com/Rokid/ShadowNode/blob/master/deps/libtuv/include/uv.h#L492, and use macros to simplify here.